tokio/src/runtime/builder.rs

```diff
@@ -490,18 +486,16 @@ cfg_rt_multi_thread! {
             use crate::loom::sys::num_cpus;
             use crate::runtime::{Kind, ThreadPool};
             use crate::runtime::park::Parker;
-            use std::cmp;
 
-            let core_threads = self.worker_threads.unwrap_or_else(|| cmp::min(self.max_threads, num_cpus()));
-            assert!(core_threads <= self.max_threads, "Core threads number cannot be above max limit");
+            let core_threads = self.worker_threads.unwrap_or_else(|| num_cpus());
 
             let (driver, resources) = driver::Driver::new(self.get_cfg())?;
 
             let (scheduler, launch) = ThreadPool::new(core_threads, Parker::new(driver));
             let spawner = Spawner::ThreadPool(scheduler.spawner().clone());
 
             // Create the blocking pool
-            let blocking_pool = blocking::create_blocking_pool(self, self.max_threads);
+            let blocking_pool = blocking::create_blocking_pool(self, self.max_blocking_threads);
```

Comment:
It seems like we need an additional change to the blocking pool. Maybe you need to add the number of worker threads to this one, or change the blocking pool itself?

Reply:
I don't see anywhere where we do `worker_threads - max_threads` to get the cap. Maybe @carllerche can chime in here?

Reply:
This should be `worker_threads + max_blocking_threads` I think.

Reply:
Eventually, we will create a special path for worker calls to spawn_blocking, but we don't have that yet.